lib/src/source/cached.dart

```diff
@@ -67,8 +67,14 @@ abstract class CachedSource extends BoundSource {
   /// Reinstalls all packages that have been previously installed into the
   /// system cache by this source.
   ///
-  /// Returns a [Pair] whose first element is the packages that were
-  /// successfully repaired and the second is the packages that failed to be
-  /// repaired.
-  Future<Pair<List<PackageId>, List<PackageId>>> repairCachedPackages();
+  /// Returns a list of results indicating for each if that package was
+  /// successfully repaired.
+  Future<Iterable<RepairResult>> repairCachedPackages();
+}
+
+/// The result of repairing a single cache entry.
+class RepairResult {
+  final bool success;
```

Comment:
Can we give this a comment, saying that this is whether or not the repair was successful.

Also maybe explain in comment what it means when repair for a package failed, is it now missing from cache?

Reply:
Done